chiselFrontend/src/main/scala/chisel3/core/Mem.scala

```diff
@@ -30,16 +30,30 @@ object Mem {
   }
 }
 
-sealed abstract class MemBase[T <: Data](t: T, val length: Int) extends HasId with VecLike[T] {
+/* The following definition of ChiselVecLike is to work around the issue that you cannot
```

Comment:
So what we've done in this cases like this is to have the implementation (do_apply) in a separate function which is abstract, and the macro transform defined at the abstract class. Since the macro only does a naming change (without resolving which subclass / etc) and the type signatures of both the macro method and function implementation are the same, the same macro should work in all subclasses.

So basically I would define the apply macro in VecLike. Vec's apply method may need to be changed to take compile options it ultimately doesn't need, but that should be fine.